#!/usr/bin/env python3
PACKAGE = "hrr_controllers"
 
from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator
 
gen = ParameterGenerator()

# PI force - gains
gen.add("K_P_fx", "double",  0, "P-Gain in force x", 1e-4,  0.0, 1e-3)
gen.add("K_P_fy", "double",  0, "P-Gain in force y", 1e-4,  0.0, 1e-3)
gen.add("K_P_fz", "double",  0, "P-Gain in force z", 1e-4,  0.0, 1e-3)
gen.add("K_I_fx", "double",  0, "I-Gain in force x", 0.0,  0.0, 1e-5)
gen.add("K_I_fy", "double",  0, "I-Gain in force y", 0.0,  0.0, 1e-5)
gen.add("K_I_fz", "double",  0, "I-Gain in force z", 0.0,  0.0, 1e-5)

# PI torque - gains
gen.add("K_P_tx", "double",  0, "P-Gain in torque x", 5e-4,  0.0, 1e-2)
gen.add("K_P_ty", "double",  0, "P-Gain in torque y", 5e-4,  0.0, 1e-2)
gen.add("K_P_tz", "double",  0, "P-Gain in torque z", 5e-4,  0.0, 1e-2)
gen.add("K_I_tx", "double",  0, "I-Gain in torque x", 0.0,  0.0, 1e-4)
gen.add("K_I_ty", "double",  0, "I-Gain in torque y", 0.0,  0.0, 1e-4)
gen.add("K_I_tz", "double",  0, "I-Gain in torque z", 0.0,  0.0, 1e-4)

exit(gen.generate(PACKAGE, "hrr_controllers", "Gains"))
